File: framework/languages/node/e2e/RegistrationCodec/Client/Scenarios/RcB3MessagePackCodecScenario.ts
// RC-B3: Root에 등록한 MessagePack extension을 사용한다 시나리오를 검증한다.
import type { CodecScenarioRes } from '../../Shared/messages';
import { postJson } from '../../../http-client';
import { ensure } from '../Support/scenario-assert';

// RC-B3 verifies MessagePack content-type selection on the main host's global codec registry.
export async function runRcB3(serverUrl: string): Promise<void> {
  const reply = await postJson<CodecScenarioRes>(serverUrl, '/codec/msgpack');
  ensure(reply.value === 'echo:rc-b3', 'RC-B3 reply value mismatch.');
  ensure(reply.contentType === 'application/x-msgpack', 'RC-B3 expected MessagePack content type.');
  const evidence = await postJson<readonly string[]>(serverUrl, '/evidence/wait', {
    containsAll: ['codec=msgpack', 'rc-b3-send', 'codec-reply|codec=msgpack'],
    timeoutMilliseconds: 10_000
  });
  ensure(evidence.some((line) => line.includes('codec-request|codec=msgpack')), 'RC-B3 request evidence missing.');
  ensure(evidence.some((line) => line.includes('codec-command|codec=msgpack')), 'RC-B3 command evidence missing.');
  console.log('scenario RC-B3 passed');
}

File: framework/languages/dotnet/e2e/RegistrationCodec/Client/Scenarios/RcB3MessagePackCodecScenario.cs
// Verifies RC-B3 Message Pack Codec behavior.
using RegistrationCodec.Client.Support;
using Zlink.HttpClient;

namespace RegistrationCodec.Client.Scenarios;

// RC-B3 verifies MessagePack codec round trip.
internal static class RcB3MessagePackCodecScenario
{
    public static async Task RunAsync(ZLinkHttpClient requester)
    {
        var result = (await requester.Post("/codec/roundtrip").Async<CodecScenarioRes>()).Body;
        ZlinkStreamAssert.Ensure(result.MessagePackValue.Contains("echo:rc-b3", StringComparison.Ordinal),
            "RC-B3 MessagePack reply mismatch.");
        ZlinkStreamAssert.Ensure(
            result.MessagePackValue.Contains("content:application/x-msgpack", StringComparison.Ordinal),
            "RC-B3 MessagePack content type mismatch.");

        Console.WriteLine("scenario RC-B3 passed");
    }
}
